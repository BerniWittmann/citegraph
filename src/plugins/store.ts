import Vue from 'vue'
import Vuex from 'vuex'
import store from '@/store'

Vue.use(Vuex)

export default store
