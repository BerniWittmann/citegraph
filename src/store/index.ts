import { RootState } from '@/store/types'
import { projects } from './modules/projects'
import { toasts } from './modules/toasts'
import { ActionTree, GetterTree, MutationTree } from 'vuex'

export const modules = {
  projects,
  toasts
}

export const state: RootState = {}

export const actions: ActionTree<RootState, RootState> = {}

export const getters: GetterTree<RootState, RootState> = {}

export const mutations: MutationTree<RootState> = {}
