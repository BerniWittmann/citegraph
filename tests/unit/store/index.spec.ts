import * as store from '@/store'

describe('store/index', () => {
  it('is an object', () => {
    expect(typeof store).toBe('object')
    expect(Object.keys(store)).toEqual([
      'modules',
      'state',
      'mutations',
      'actions',
      'getters'
    ])
  })
  it('returns state', () => {
    expect(store.state).toMatchSnapshot()
  })
  it('returns actions', () => {
    expect(store.actions).toMatchSnapshot()
  })
  it('returns mutations', () => {
    expect(store.mutations).toMatchSnapshot()
  })
  it('returns getters', () => {
    expect(store.getters).toMatchSnapshot()
  })
  it('returns modules', () => {
    expect(store.modules).toMatchSnapshot()
  })
})
