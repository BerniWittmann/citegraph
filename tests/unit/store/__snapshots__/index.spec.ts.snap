// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store/index returns actions 1`] = `Object {}`;

exports[`store/index returns getters 1`] = `Object {}`;

exports[`store/index returns modules 1`] = `Object {}`;

exports[`store/index returns mutations 1`] = `Object {}`;

exports[`store/index returns state 1`] = `Object {}`;
